### DISCUSSION TITLE
Github Greatest Hits

### DISCUSSION THREAD

**Discussion by philipperemy** [community/community#38452] [Other Feature Feedback, Questions, & Ideas] [answered]
Hey,

Does someone know where I can find the list of the repositories that are part of the Github Greatest Hits?

I saw it somewhere long time ago but I can't find it again.

Here is the link to the greatest hits project: 

- https://archiveprogram.github.com/greatest-hits/
- https://www.bestdevops.com/github-shares-greatest-hits-open-source-repositories-with-historic-libraries/
- https://siliconangle.com/2020/11/19/github-archives-open-source-greatest-hits-four-locations-around-world/

PS: I'm not talking about the Arctic Vault. Just the "top" 17,000 projects.

Thank you in advance,

Philippe

**Comment by ws-jm** [comment]
This will give you all the user's public repositories. If you need to find out private repositories you will need to authenticate as the particular user. You can then use the REST call:
To do this in Python do something like:

USER='AUSER'
API_TOKEN='ATOKEN'
GIT_API_URL='https://api.github.com'

def get_api(url):
    try:
        request = urllib2.Request(GIT_API_URL + url)
        base64string = base64.encodestring('%s/token:%s' % (USER, API_TOKEN)).replace('\n', '')
        request.add_header("Authorization", "Basic %s" % base64string)
        result = urllib2.urlopen(request)
        result.close()
    except:
        print 'Failed to get api request from %s' % url

**Comment by ws-jm** [comment]
This will only give the first "page" of the result set, which is set at 30 items by default. You can use ?per_page=100 to get the maximum ammount but if a user has more than a hundred repos you will need to follow several next URLs in the HTTP Link header send with the response

**Comment by philipperemy** [comment]
@ws-jm I'm not talking about getting the user's public repositories. Re-read my message please. Thanks!

**Comment by ws-jm** [comment]
so what do you want??

**Reply by philipperemy** [reply]
@ws-jm man just re-read my post you will understand.

**Comment by ws-jm** [comment]
ok

**Comment by philipperemy** [comment, marked as answer]
For info it was here: https://archiveprogram.github.com/assets/img/archive-repos.txt

**Comment by lawcompanion1** [comment]
Hello.
